langchain/src/chains/base.ts

```diff
@@ -9,6 +9,7 @@ import type {
   ChatQAChain,
 } from "./index.js";
 import { BaseMemory } from "../memory/index.js";
+import { ChatModelChain } from "./index.js";
```

Comment:
This should be moved to the `import type` statement above (to avoid circular deps, which are allowed for type definitions but less allowed for actual code